pkg/registry/consul.go

```diff
@@ -110,27 +108,20 @@ func (c *ConsulClient) Release(sessionID string, key string, value string) bool
 	return isReleased
 }
 
-// Watch watches a key
-func (c *ConsulClient) Watch(watchType string, key string, handler HandlerFunc) error {
+// Watch watches a key and returns a watcher instance, this instance should be used to terminate the watch
+func (c *ConsulClient) Watch(wh *WatchConfig) (IWatcher, error) {
 	plan, err := watch.Parse(map[string]interface{}{
-		"type":   watchType,
-		"prefix": key,
+		"type":   wh.WatchType,
+		"prefix": wh.WatchPath,
 	})
 
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	wh := NewConsulWatchHandler(handler)
+	cwh := NewConsulWatcher(wh, plan, c.client)
```

Comment:
can IWatcher interface be injected in this func. Would help with unit testing

Reply:
This func is being used to create watcher, method can be renamed but not sure if interface injection is possible, we will need to create interface in registry